### DISCUSSION TITLE
Support for a 'Stacked Diff' workflow

### DISCUSSION THREAD

**Discussion by kvikas** [magit/forge#593] [Ideas]
* [This post](https://jg.gg/2018/09/29/stacked-diffs-versus-pull-requests/) explains a stacked diff workflow in more detail.
* There are tools to mimic this workflow on Github: [SPR](https://github.com/ejoffe/spr) and well [another SPR](https://github.com/getcord/spr)

The idea is to support this workflow in Github forge.

**Comment by tarsius** [comment]
I am skeptical of this workflow. I feel like I am already doing this myself. Except for the one-pr-per-commit, which is of course core to this workflow, so that feeling should be taken with a grain of salt. I also think that Magit already supports to an extend, by making rebasing simpler, and that a somewhat recent change to Git also makes it less painful to maintain a stack of linear branches/pull-requests: if you rebase the branch that contains all the others, then those will now be moved forward too when you rebase.

After a quick look at the first tool, I am guessing you want additional information to be displayed about the state of the various pull-requests. That is reasonable, but could also be implemented in a separate package (though I might have to provide a hook or two, to make it easier to inject the additional information). Also note that Forge currently doesn't even display CI status information about individual pull-requests. Until that is available, it does not make sense to implement status information for more specialized things.

This workflow does not appeal to me because for the kinds of projects I contribute to, and the kinds of teams I work with, it would be overkill. Heck, we usually don't even do pull-requests. As a consequence I am not the right person to design and implement support for this workflow.

IMO this should be implemented as a separate package.

You might also be interested in https://github.com/stacked-git/magit-stgit.
